File: start_server.py
#!/usr/bin/env python3
"""
Simple HTTP server for OliverSEO Dashboard
Solves CORS issues when viewing markdown files
"""

import http.server
import socketserver
import os
import webbrowser
from pathlib import Path

# Get the directory where this script is located
SCRIPT_DIR = Path(__file__).parent.absolute()
PORT = 8000

class CORSRequestHandler(http.server.SimpleHTTPRequestHandler):
    """HTTP Request Handler with CORS support"""
    
    def end_headers(self):
        # Add CORS headers
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', '*')
        super().end_headers()
    
    def do_OPTIONS(self):
        """Handle OPTIONS request for CORS preflight"""
        self.send_response(200)
        self.end_headers()
    
    def log_message(self, format, *args):
        """Custom log format"""
        print(f"[{self.address_string()}] {format % args}")

def main():
    # Change to the script directory
    os.chdir(SCRIPT_DIR)
    
    print("=" * 60)
    print("🚀 OliverSEO Local Server")
    print("=" * 60)
    print(f"📁 Serving from: {SCRIPT_DIR}")
    print(f"🌐 Server running at: http://localhost:{PORT}")
    print(f"📄 Dashboard: http://localhost:{PORT}/dashboard-fixed.html")
    print("=" * 60)
    print("\n💡 Press Ctrl+C to stop the server\n")
    
    # Create server
    with socketserver.TCPServer(("", PORT), CORSRequestHandler) as httpd:
        try:
            # Try to open browser automatically
            try:
                webbrowser.open(f'http://localhost:{PORT}/dashboard-fixed.html')
                print("✅ Browser opened automatically!")
            except:
                print("⚠️  Could not open browser automatically")
                print(f"   Please open: http://localhost:{PORT}/dashboard-fixed.html")
            
            print("\n🔄 Server is running...\n")
            httpd.serve_forever()
        except KeyboardInterrupt:
            print("\n\n🛑 Server stopped by user")
            print("👋 Goodbye!")

if __name__ == "__main__":
    main()

File: start_server.js
/**
 * Simple HTTP server for OliverSEO Dashboard
 * Solves CORS issues when viewing markdown files
 * 
 * Run with: node start_server.js
 */

const http = require('http');
const fs = require('fs');
const path = require('path');
const { exec } = require('child_process');

const PORT = 8000;
const DIRECTORY = __dirname;

// MIME types
const mimeTypes = {
    '.html': 'text/html',
    '.js': 'text/javascript',
    '.css': 'text/css',
    '.json': 'application/json',
    '.png': 'image/png',
    '.jpg': 'image/jpg',
    '.gif': 'image/gif',
    '.svg': 'image/svg+xml',
    '.ico': 'image/x-icon',
    '.md': 'text/markdown',
    '.txt': 'text/plain'
};

const server = http.createServer((req, res) => {
    console.log(`${req.method} ${req.url}`);

    // CORS headers
    res.setHeader('Access-Control-Allow-Origin', '*');
    res.setHeader('Access-Control-Allow-Methods', 'GET, OPTIONS');
    res.setHeader('Access-Control-Allow-Headers', '*');

    if (req.method === 'OPTIONS') {
        res.writeHead(200);
        res.end();
        return;
    }

    // Parse URL
    let filePath = '.' + req.url;
    if (filePath === './') {
        filePath = './dashboard-fixed.html';
    }

    // Get file extension
    const extname = String(path.extname(filePath)).toLowerCase();
    const contentType = mimeTypes[extname] || 'application/octet-stream';

    // Read file
    fs.readFile(path.join(DIRECTORY, filePath), (error, content) => {
        if (error) {
            if (error.code === 'ENOENT') {
                res.writeHead(404, { 'Content-Type': 'text/html' });
                res.end('<h1>404 - File Not Found</h1>', 'utf-8');
            } else {
                res.writeHead(500);
                res.end(`Server Error: ${error.code}`, 'utf-8');
            }
        } else {
            res.writeHead(200, { 'Content-Type': contentType });
            res.end(content, 'utf-8');
        }
    });
});

server.listen(PORT, () => {
    console.log('='.repeat(60));
    console.log('🚀 OliverSEO Local Server');
    console.log('='.repeat(60));
    console.log(`📁 Serving from: ${DIRECTORY}`);
    console.log(`🌐 Server running at: http://localhost:${PORT}`);
    console.log(`📄 Dashboard: http://localhost:${PORT}/dashboard-fixed.html`);
    console.log('='.repeat(60));
    console.log('\n💡 Press Ctrl+C to stop the server\n');

    // Try to open browser
    const url = `http://localhost:${PORT}/dashboard-fixed.html`;
    const command = process.platform === 'win32' 
        ? `start ${url}` 
        : process.platform === 'darwin' 
        ? `open ${url}` 
        : `xdg-open ${url}`;

    exec(command, (error) => {
        if (error) {
            console.log('⚠️  Could not open browser automatically');
            console.log(`   Please open: ${url}`);
        } else {
            console.log('✅ Browser opened automatically!');
        }
    });

    console.log('🔄 Server is running...\n');
});


